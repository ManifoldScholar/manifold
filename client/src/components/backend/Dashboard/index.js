import Notifications from './Notifications.js';

export default {
  Notifications
};
