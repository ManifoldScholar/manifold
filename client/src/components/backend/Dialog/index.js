import Wrapper from "./Wrapper";
import Confirm from "./Confirm";
import ResetPassword from "./ResetPassword";

export default {
  Wrapper,
  Confirm,
  ResetPassword
};
