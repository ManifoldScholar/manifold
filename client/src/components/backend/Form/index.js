import HasMany from './HasMany';
import Radios from './Radios';
import TextArea from './TextArea';
import TextInput from './TextInput';
import MaskedTextInput from './MaskedTextInput';
import Upload from './Upload';
import Select from './Select';
import Save from './Save';
import Switch from './Switch';
import Date from './Date';
import HigherOrder from './HigherOrder';
import Hidden from './Hidden';
import AvatarBuilder from './AvatarBuilder/AvatarBuilder';
import setter from './setter';
import FieldGroup from './FieldGroup';

export default {
  HasMany,
  Hidden,
  Radios,
  TextArea,
  TextInput,
  MaskedTextInput,
  Upload,
  Select,
  Save,
  Switch,
  Date,
  AvatarBuilder,
  HigherOrder,
  FieldGroup,
  setter
};
