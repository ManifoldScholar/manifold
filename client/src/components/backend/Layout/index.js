import Header from './Header.js';
import PanelNav from './PanelNav';

export default {
  Header,
  PanelNav
};
