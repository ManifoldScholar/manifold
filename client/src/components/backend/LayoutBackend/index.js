import Header from './Header.js';

export default {
  Header
};
