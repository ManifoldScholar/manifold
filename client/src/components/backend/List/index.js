import SimpleList from "./SimpleList";
import Searchable from "./Searchable";

export default {
  SimpleList,
  Searchable
};
