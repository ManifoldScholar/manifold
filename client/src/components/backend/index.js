export Dashboard from './Dashboard';
export Form from './Form';
export Layout from './Layout';
export Project from './Project';
export List from './List';
export Resource from './Resource';
export User from './User';
export Drawer from './Drawer';
export Dialog from './Dialog';
export Navigation from './Navigation';
export Category from './Category';
export Maker from './Maker';
export Event from './Event';
