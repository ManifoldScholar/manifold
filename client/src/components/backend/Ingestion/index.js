import Form from "./Form";
import ConnectionError from "./ConnectionError";

export default {
  Form,
  ConnectionError
};
