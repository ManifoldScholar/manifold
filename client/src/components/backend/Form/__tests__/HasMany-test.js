import React from 'react';
import renderer from 'react-test-renderer';
import { Form } from 'components/backend';
import { Provider } from 'react-redux';
import createStore from 'store/createStore';

describe("Backend.Form.HasMany component", () => {

  const store = createStore();

  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store}>
        <Form.HasMany
          label="A form label"
          entities={[
            { id: 1, attributes: { name: "John"}},
            { id: 2, attributes: { name: "Jane"}}
          ]}
          onChange={() => {}}
          optionsFetch={() => {}}
          onNew={() => {}}
          entityLabelAttribute="name"
          orderable
        />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
