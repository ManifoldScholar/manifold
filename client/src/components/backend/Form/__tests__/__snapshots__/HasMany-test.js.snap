// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Backend.Form.HasMany component renders correctly 1`] = `
<div
  className="form-input"
  style={Object {}}
>
  <h4
    className="form-input-heading"
  >
    makers
  </h4>
  <nav
    className="has-many-list"
  >
    <ul>
      <li>
        <div
          className="association"
        >
          <h4
            className="association-name"
          >
            John
          </h4>
        </div>
        <div
          className="utility"
        >
          <button
            className="manicon manicon-arrow-down"
            onClick={[Function]}
          >
            <span
              className="screen-reader-text"
            >
              Move John down in the makers list
            </span>
          </button>
          <button
            className="manicon manicon-x"
            onClick={[Function]}
          >
            <span
              className="screen-reader-text"
            >
              Remove 
              John
               from the 
              makers
               list.
            </span>
          </button>
        </div>
      </li>
      <li>
        <div
          className="association"
        >
          <h4
            className="association-name"
          >
            Jane
          </h4>
        </div>
        <div
          className="utility"
        >
          <button
            className="manicon manicon-arrow-up"
            onClick={[Function]}
          >
            <span
              className="screen-reader-text"
            >
              Move Jane up in the makers list
            </span>
          </button>
          <button
            className="manicon manicon-x"
            onClick={[Function]}
          >
            <span
              className="screen-reader-text"
            >
              Remove 
              Jane
               from the 
              makers
               list.
            </span>
          </button>
        </div>
      </li>
    </ul>
    <div
      className="input-predictive"
    >
      <div
        className="input"
      >
        <i
          aria-hidden="true"
          className="manicon manicon-plus"
        />
        <input
          aria-describedby={null}
          aria-label={undefined}
          className="text-input"
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          placeholder={undefined}
          type="text"
          value=""
        />
        <button
          className="submit"
          onClick={[Function]}
          type="button"
        >
          Create New
        </button>
      </div>
      <nav
        className="predictive-list"
      >
        <ul />
      </nav>
    </div>
  </nav>
</div>
`;
